server/src/main/java/io/spine/server/event/IntegrationEventConverter.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2018, TeamDev. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.server.event;
+
+import com.google.common.base.Converter;
+import com.google.protobuf.Any;
+import com.google.protobuf.Timestamp;
+import io.spine.core.Event;
+import io.spine.core.EventContext;
+import io.spine.server.integration.IntegrationEvent;
+import io.spine.server.integration.IntegrationEventContext;
+
+import java.io.Serializable;
+
+import static io.spine.protobuf.TypeConverter.toAny;
+import static io.spine.util.Exceptions.unsupported;
+
+/**
+ * Converts {@code IntegrationEvent} into {@code Event}.
+ *
+ * <p>Does not support backward conversion.
```

Comment:
How come it's a `Converter` then?

Reply:
Guava's `Converter` does allow to throw in `doBackward()`.

Reply:
See the issue for making it a read `Converter` (#777).